### ISSUE TITLE
Multi Temporal Datamodule not loading data properly

### ISSUE THREAD

**Issue by Lrogic** [terrastackai/terratorch#706]
**Describe the issue**
When loading multitemporal image data the resulting tensor is not in the expected format. The returned shape of the tensor for the **MultiTemporalCropClassificationDataModule** datamodule is in the shape (6, 3, 224, 224). I would expect the first axis to correspond to the 6 bands of the image, and the second axis to correspond to the timesteps of the image in order. But when you look at the image tensor returned by the datamodule this is not what's returned.

**To Reproduce (optional, but appreciated)**
Run the [ `prithvi_v2_eo_300_tl_unet_multitemporal_crop.ipynb`](https://github.com/IBM/terratorch/blob/main/examples/tutorials/PrithviEOv2/prithvi_v2_eo_300_tl_unet_multitemporal_crop.ipynb) tutorial until you've created the datamodule.

Then run the following code:
```
# data from the original tif
with rasterio.open('/content/multi-temporal-crop-classification-subset/training_chips/chip_003_062_merged.tif') as src:
  original_data = src.read()

print("original data shape: ", original_data.shape)
# get the first image of the time series
original_first_img = original_data[:6]
print("original first image shape: ", original_first_img.shape)

# data after being transformed by the datamodule
train_dataset = datamodule.train_dataset
datamodule_data = train_dataset[0]['image'].detach().cpu().numpy()
print("data loaded from datamodule shape: ", datamodule_data.shape)
# get the first image from the time series
datamodule_first_img = datamodule_data[:, 0]
print("first image loaded from datamodule shape: ", datamodule_first_img.shape)

# we should expect these to be the same
res = np.allclose(original_first_img, datamodule_first_img, atol=1e-3)
print("Are the images the same: ", res)

# look at some of the data
print(original_first_img)
print('-------------------------')
print(datamodule_first_img)
```

**Screenshots or log output (optional)**
![Image](https://github.com/user-attachments/assets/1d74f32d-835c-423d-9fc6-589b41f51277)

![Image](https://github.com/user-attachments/assets/b1985f93-09ef-4f7c-944c-fc52a6904c3c)

As can be seen in the output, just the first band of the first image matches, then after that the fourth band of the original matches with the second band of the datamodule loaded image, after that the bands do not match.

**Expected behavior (optional)**
I would expect that retrieving the first timestep image from the datamodule loaded data would match the first timestep image in the original data. I might be misunderstanding how the data is transformed/loaded, so please let me know if this is not a bug and I'm not understanding something.

**Deployment information (optional)**
 - TerraTorch version: 1.0.1

**Comment by Joao-L-S-Almeida** [comment]
The files contained in the dataset have 18 bands and the dataset is reshaping them
```
        # to channels last
        image = image.to_numpy()
        if self.expand_temporal_dimension:
            image = rearrange(image, "(channels time) h w -> channels time h w", channels=len(self.bands))
        image = np.moveaxis(image, 0, -1)

```
in order to create the time dimension, but it sounds strange to me, since it isn't clear that those bands represent multiple timesteps stored in the same file. 
Any comment @romeokienzler @blumenstiel ?

**Comment by FahmiGozal** [comment]
@Lrogic My hypothesis would be the "*_merged.tif" of size (18 x 224 x 224) is stacked in the (Temporal x Channel) Dimension as (B_1, B_2, B_3, G_1, G_2, G_3, ... , SWIR2_1, SWIR_2, SWIR_3) as when i tried to observe the value of the unstacked channels from **MultiTemporalCropClassificationDataModule** after editing the value for band indices 12 to 17 to 0s, the fifth and sixth channel is all zeroes

**Comment by blumenstiel** [comment]
I used the data once and it worked but did not looked to much into it. @paolofraccaro @daniszw any comments?

**Comment by Lrogic** [comment]
@FahmiGozal I think you are right, I just did a short script to check. Is that the expectation for all multi temporal files when loading them in Terratorch? 

You should stack as: B1, B2, B3, G1, G2, G3, ...

And not as: (all bands from IMG1), (all bands from IMG2), (all bands from IMG3)

If it is maybe there could be some documentation clarifying the correct setup, because I don't think I saw it anywhere previously. Thanks

**Comment by romeokienzler** [comment]
potential duplicate of #768
